File: Application/Services/Operations/StockProduct/Dtos/StockDto.cs
using System.Collections.Generic;
using Application.Services.Operations.StockProduct.ProductKind;
using Application.Services.Shared.Dtos;

namespace Application.Services.Operations.StockProduct
{
    public class StockDto: RootBaseDto
    {
        // public ProductDto Product { get; set; }
        public ProductDto Product { get; set; }
        public List<ItemProductDto> ItemsProducts { get; set; }
       // public List<Sold> Solds { get; set; }
    }
}

File: Views/src/components/stock-product/product/dtos/stock-dto.ts
import { RootBase } from "src/shared/entities-dtos/root-base";
import { SegmentDto } from "./segment-dto";
import { ProductDto } from "./product-dto";
import { ItemProductDto } from "./item-product";

export class StockDto extends RootBase {
     ProductDto: ProductDto;
     ItemsProducts:ItemProductDto[];
}